frontend/packages/insights-plugin/src/plugin.tsx

```diff
@@ -0,0 +1,35 @@
+import {
+  Plugin,
+  DashboardsOverviewHealthPrometheusSubsystem,
+  ModelFeatureFlag,
+} from '@console/plugin-sdk';
+import { ClusterVersionModel } from '@console/internal/models';
+import { referenceForModel } from '@console/internal/module/k8s';
+import { getClusterInsightsStatus } from './components/InsightsPopup/status';
+
+type ConsumedExtensions = ModelFeatureFlag | DashboardsOverviewHealthPrometheusSubsystem;
+
+const plugin: Plugin<ConsumedExtensions> = [
+  {
+    type: 'Dashboards/Overview/Health/Prometheus',
+    properties: {
+      title: 'Insights',
+      queries: ["health_statuses_insights{metric=~'low|moderate|important|critical'}"],
+      healthHandler: getClusterInsightsStatus,
+      additionalResource: {
+        kind: referenceForModel(ClusterVersionModel),
+        namespaced: false,
+        name: 'version',
+        isList: false,
+        prop: 'cluster',
+      },
+      popupComponent: () =>
+        import('./components/InsightsPopup/index' /* webpackChunkName: "insights-plugin" */).then(
+          (m) => m.InsightsPopup,
+        ),
+      popupTitle: 'Insights status',
+    },
```

Comment:
You need to gate the extension by some model contributed by Insights operator. To do so:

1. create ModelDefinition extension
```js
{
    type: 'ModelDefinition',
    properties: {
      models: [YourModel],
    },
  },
```

2. Add FeatureFlag extension
```js
const INSIGHTS_FLAG = 'INSIGHTS_FLAG';
```
```js
{
    type: 'FeatureFlag/Model',
    properties: {
      model: YourModel,
      flag: INSIGHTS_FLAG,
    },
  },
```
 
3. gate Dashboard Health with the above flag

```suggestion
   flags: {
      required: [InsightsFlag],
    },
    },
```

Reply:
@bond95 As @rawagner wrote, we should enable the plugin's extensions only when the cluster has the right capability (i.e. Insights operator is installed).

Detecting Insights operator can be done via CRD detection. Use `ModelDefinition` extension to make Console aware of your CRD, then use `FeatureFlag/Model` extension to introduce a feature flag reflecting the presence of such CRD.

Every other extension can then use the `flags` object to declare which feature flags are required and/or disallowed in order for that particular extension to be in use. Typically, you'd want all Insights extensions to be gated by `INSIGHTS_FLAG`.

Reply:
@rawagner @vojtechszocs What if we don't have any CRD?

Reply:
Another option is to implement your own detection logic and use `CustomFeatureFlag` . Example https://github.com/openshift/console/blob/master/frontend/packages/ceph-storage-plugin/src/plugin.ts#L76

Reply:
@rawagner We had discussion with @smarterclayton and @spadgett, they suggested not to hide tab even if insights-operator is disabled.

Reply:
@bond95 Assuming that Insights integration is implemented via cluster-level operator (i.e. not via OLM operator), it makes sense to have Insights plugin extensions active by default :+1: